pkg/flags/client.go

```diff
@@ -17,52 +21,36 @@ type Config struct {
 	Timeout time.Duration
 }
 
-// FlagValues is the struct for our keys and values from LD
-type FlagValues map[string]interface{}
-
-// FlagClient provides a way to retrieve the set of flags
-type FlagClient interface {
-	Flags(user lduser.User) FlagValues
-}
-
-// LaunchDarklyClient is backed by the LaunchDarkly service and requires
-// network access.
-type LaunchDarklyClient struct {
-	client *ld.LDClient
-}
-
-// Flags returns the flags from Launch Darkly
-func (c LaunchDarklyClient) Flags(user lduser.User) FlagValues {
-	currentFlags := FlagValues{}
-	state := c.client.AllFlagsState(user, flagstate.OptionClientSideOnly())
-	valuesMap := state.ToValuesMap()
-	for k, v := range valuesMap {
-		currentFlags[k] = v
+// NewLaunchDarklyClient returns a client backed by Launch Darkly
+func NewLaunchDarklyClient(config Config) (*ld.LDClient, error) {
+	if config.Source == appconfig.FlagSourceLaunchDarkly {
+		return ld.MakeClient(config.Key, config.Timeout)
 	}
-	return currentFlags
-}
 
-// LocalFlagClient returns a set of flags specified locally
-type LocalFlagClient struct {
-	flags FlagValues
+	// we default to an OFFLINE client for non-deployed environments
+	return ld.MakeCustomClient("fake_offline_key", ld.Config{Offline: true}, 5*time.Second)
 }
 
-// Flags returns all flags
-func (c LocalFlagClient) Flags(user lduser.User) FlagValues {
-	return c.flags
-}
+// Principal builds the LaunchDarkly user object for the
+// currently authenticated principal.
+func Principal(ctx context.Context) lduser.User {
+	p := appcontext.Principal(ctx)
 
-// NewLaunchDarklyClient returns a client backed by Launch Darkly
-func NewLaunchDarklyClient(config Config) (*ld.LDClient, error) {
-	return ld.MakeClient(config.Key, config.Timeout)
-}
+	// we should not be using bare EUA IDs as identifiers to
+	// LaunchDarkly (per Jimil/ISSO), so we use a cryptographically
+	// secure one-way hash of the EUA ID as "key" for the LD User object.
+	h := sha256.New()
+	_, _ = h.Write([]byte(p.ID()))
+	key := fmt.Sprintf("%x", h.Sum(nil))
 
-// WrapLaunchDarklyClient returns a client that fulfils the flags.FlagClient interface
-func WrapLaunchDarklyClient(c *ld.LDClient) *LaunchDarklyClient {
-	return &LaunchDarklyClient{client: c}
-}
+	// this is a bit of a loose inference, assuming a user w/o Job Codes
+	// is an Anonymous user. Over time, may want to consider adding
+	// a `func Anonymous() bool` accessor to the authn.Principal interface
+	// definition instead of doing this inference
+	authed := (p.AllowEASi() || p.AllowGRT())
```

Comment:
Is it possible to have AllowGRT _without_ having AllowEASi? If not, this line could just be authed := p.AllowEASi

Reply:
Good eye!
That is _functionally_ true in practice, but not _structurally_ true... If you look at the reviewer/"BOSS" case in the unit tests, that is one way an `EUAPrincipal` can be constructed w/o the `AllowEASi`. 
I am open to plugging that hole in the future, especially once we start bringing in more Job Codes (like for 508 stuff).